Вопросы по курсовой

e-andryushina opened the pull request:
Вопросы по курсовой

e-andryushina committed:
Вопросы по курсовой

Bl3cster commented on src/Employee.java:
```diff
@@ -0,0 +1,63 @@
+public class Employee {
+
+    private static int countEmployeeID = 1;
+    private final int employeeID;
```
уже не сработает, у тебя не может меняться id, так как он final... тогда возникает вопрос: зачем он нужен?!

Bl3cster commented on src/Employee.java:
```diff
@@ -0,0 +1,63 @@
+
+    private static int countEmployeeID = 1;
+    private final int employeeID;
+    private final String employee;
```
здесь ты утверждаешь, что ссылка будет постоянная, что то же не есть хорошо. убирай final ото всюду. в этих полях он не нужен

Bl3cster commented on src/Employee.java:
```diff
@@ -0,0 +1,63 @@
+
+
+    public Employee (String employee, String businessUnit, double salary) {
+        this.employeeID = countEmployeeID++;
```
как же мы его поменяем?

Bl3cster commented on src/EmployeeReport.java:
```diff
@@ -0,0 +1,118 @@
+    private String employeeMaxSalary;
+    private String employeeMinSalary;
+
+    public EmployeeReport (int rSize) {
```
на что указывает r?

Bl3cster commented on src/EmployeeReport.java:
```diff
@@ -0,0 +1,118 @@
+    }
+
+
+    public static int  sizeCount() {
```
чем не устроил employees.length?

e-andryushina replied:
> чем не устроил employees.length?

employees.length дает общее кол-во ячеек в массиве. Но, чтобы рассчитать среднее арифм. самым понятным для меня способом, я нахожу фактическое количество заполненных ячеек массива. Т.к. если массив заполнен неполностью,  ср. ар. при использовании employees.length рассчитается неправильно. По крайней мере, я так поняла. Возможно, это некорректный способ.

Bl3cster commented on src/EmployeeReport.java:
```diff
@@ -0,0 +1,118 @@
+                salaryMonthTotal += employee.getSalary();
+            }
+        }
+        System.out.println("Сумма затрат на зарплаты отдела за месяц составляет " + salaryMonthTotal + " рублей.");
```
а где в этом методе хоть слово про отдел?

Bl3cster commented on src/EmployeeReport.java:
```diff
@@ -0,0 +1,118 @@
+            }
+        }
+        System.out.println("У работника " + employeeMaxSalary +
+                " максимальная зарплата в отделе - " + salaryMonthMax +
```
аналогично, не вижу, что здесь по отделу сравнивают

e-andryushina committed:
Вопросы по курсовой

e-andryushina committed:
Курсовая на проверку

Bl3cster commented on src/Main.java:
```diff
@@ -1,6 +1,51 @@
+
+
     public static void main(String[] args) {
+        EmployeeReport employeeReport = new EmployeeReport();
```
зачем?

e-andryushina committed:
Внесенные изменения:
1. Модификатором private ограничен доступ к полям класса Employee
2. Добавлены доп.проверки для передаваемых значений
3. Добавлены методы: добавить работника, удалить работника, изменить з/п, подразделение
